### PR TITLE
PAINTROID-409 Update Splashscreen to new Splashscreen API(android 12)

### PR DESCRIPTION
**Pull request by Aayush2111**
jira ticket: https://jira.catrob.at/browse/PAINTROID-409

Dark mode off:
image

Dark mode on:
image

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by Aayush2111** [comment]
Hey @ThorstenBandel 

I have deleted the previous pull request due to some errors.
I have taken the time to address these issues and have created a new pull request. The new pull request includes all of the necessary fixes and improvements!

**Comment by juliajulie95** [comment]
The icon seems to be cut off in your screenshots, please have a look

**Comment by SebastianGrief** [comment]
I've tested this on a API 22 emulator and a device, both do not show the logo. 
Could you have a look at that please?

In the meantime, I'll try and find a solution for the image size problem.

**Comment by Aayush2111** [comment]
> I've tested this on a API 22 emulator and a device, both do not show the logo. Could you have a look at that please?
> 
> In the meantime, I'll try and find a solution for the image size problem.

I'll check that out again!

**Comment by Aayush2111** [comment]
> I've tested this on a API 22 emulator and a device, both do not show the logo. Could you have a look at that please?
> 
> In the meantime, I'll try and find a solution for the image size problem.

I'll check that out again!

**Comment by SebastianGrief** [comment]
> > I've tested this on a API 22 emulator and a device, both do not show the logo. Could you have a look at that please?
> > In the meantime, I'll try and find a solution for the image size problem.
> 
> It seems to be functioning properly on my end. Have you attempted to manually open the application? It's possible that if the application was installed directly from Android Studio, any updates or modifications may not be reflected.

Did you test on an emulator or do you have a physical device with API 22?

**Comment by Aayush2111** [comment]
> > > I've tested this on a API 22 emulator and a device, both do not show the logo. Could you have a look at that please?
> > > In the meantime, I'll try and find a solution for the image size problem.
> > 
> > 
> > It seems to be functioning properly on my end. Have you attempted to manually open the application? It's possible that if the application was installed directly from Android Studio, any updates or modifications may not be reflected.
> 
> > Known incompatibilities:
> > [...]
> > On API 21-22, The icon isn't displayed until the application starts, only the background is visible.

I see. What shall be done for that then?

File: Paintroid/src/main/java/org/catrobat/paintroid/MainActivity.kt [hunk 1]
```diff
@@ -279,8 +280,8 @@ class MainActivity : AppCompatActivity(), MainView, CommandListener {
     }
 
     public override fun onCreate(savedInstanceState: Bundle?) {
-        setTheme(R.style.PocketPaintTheme)
         super.onCreate(savedInstanceState)
+        installSplashScreen()
```
Comment (foenabua) [inline Paintroid/src/main/java/org/catrobat/paintroid/MainActivity.kt:284]: in the [migration guide](https://developer.android.com/develop/ui/views/launch/splash-screen/migrate#best-practices) it is noted to call installSplashScreen() before super.onCreate().
Please change this

**Review by foenabua — CHANGES_REQUESTED** [review]

File: Paintroid/build.gradle [hunk 2]
```diff
@@ -149,6 +149,8 @@ dependencies {
     androidTestImplementation "androidx.test.uiautomator:uiautomator:2.2.0"
     testImplementation "androidx.test:core-ktx:1.4.0"
     implementation 'com.android.support.test.espresso:espresso-idling-resource:3.1.0'
+
+    implementation 'androidx.core:core-splashscreen:1.0.0-beta02'
```
Comment (foenabua) [inline Paintroid/build.gradle:153]: Please change version to androidx.core:core-splashscreen:1.0.1

File: build.gradle [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
-    ext.kotlin_version = '1.4.32'
+    ext.kotlin_version = '1.6.0'
```
Comment (foenabua) [inline build.gradle:22]: Please resolve merge conflicts. You can use kotlin v1.6.0

**Review by foenabua — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 4]
```diff
@@ -1,3 +1,4 @@
 android.enableJetifier=true
 android.useAndroidX=true
-org.gradle.jvmargs=-XX:MaxPermSize=1024m -Xmx4096m
\ No newline at end of file
+org.gradle.jvmargs=-XX:MaxPermSize=1024m -Xmx4096m
+kotlin.jvm.target.validation.mode = IGNORE
```
Comment (foenabua) [inline gradle.properties:4]: I didn't understand why you added kotlin.jvm.target.validation.mode = IGNORE. Could you please explain?

**Comment by juliajulie95** [comment]
Please reopen if you are still working on this.

**Pull request closed without merging**